internal/backends/sqlite/metadata/registry.go

```diff
@@ -58,11 +71,23 @@ func (r *Registry) Close() {
 	r.p.Close()
 }
 
-// CollectionToTable converts FerretDB collection name to SQLite table name.
-func (r *Registry) CollectionToTable(collectionName string) string {
-	// TODO https://github.com/FerretDB/FerretDB/issues/2749
-	h := sha1.Sum([]byte(collectionName))
-	return hex.EncodeToString(h[:])
+// collectionToTable converts FerretDB collection name to SQLite table name.
+// It returns error if the name is incorrect for SQLite backend.
+func collectionToTable(collectionName string) (string, error) {
+	if strings.HasPrefix(collectionName, reservedPrefix) {
+		return "", ErrInvalidCollectionName
+	}
+
+	hash32 := fnv.New32a()
+	must.NotFail(hash32.Write([]byte(collectionName)))
+
+	// SQLite table cannot start with _sqlite prefix
+	if strings.HasPrefix(strings.ToLower(collectionName), reservedTablePrefix) {
+		collectionName = "_" + collectionName
+	}
+	// TODO case insensitive
```

Comment:
The hashes for `collection` and `COLLECTION` will be different, right? So, it seems that there is not much to handle here (but we might want to add tests, yes).

Reply:
They will be different, although it's not possible to test in a reasonable way. We have a lot of possibilities like fOo Foo foO... Just want to be sure that we don't expect any hash collision here, or should we create some validation just to be sure

Reply:
Ok, if sqlite table names are case insensitive, what I can suggest is to "unify" the way we call them, e.g. by lower-casing the names, so then in the meta data we store all the table names lower-cased (for simplicity). That's for the scope of this task. Collisions are not for the scope of this task. However, if we can create a unique index from the field `table_name`, let's do it, and we will be safe for now.